source/slang/slang-parameter-binding.cpp

```diff
@@ -1065,81 +1075,87 @@ static void addExplicitParameterBindings_GLSL(
     // the index/offset/etc.
     //
 
-    TypeLayout::ResourceInfo* resInfo = nullptr;
-    TypeLayout::ResourceInfo* foundResInfo = nullptr;
-
-    LayoutSemanticInfo semanticInfo;
-    semanticInfo.index = 0;
-    semanticInfo.space = 0;
+    const int kMaxResCount = 2;
+    const int kSubpassResInfo = 1;
+    const int kResInfo = 0;
 
-    LayoutSemanticInfo subpassSemanticInfo;
-    bool foundBinding = false;
-    bool foundSubpass = false;
+    TypeLayout::ResourceInfo* foundResInfo = nullptr;
+    struct ResAndSemanticInfo
+    {
+        TypeLayout::ResourceInfo* resInfo = nullptr;
+        LayoutSemanticInfo semanticInfo;
+        ResAndSemanticInfo()
+        {
+            semanticInfo.index = 0;
+            semanticInfo.space = 0;
+        }
+    };
+    ResAndSemanticInfo info[kMaxResCount] = {};
```

Comment:
I don't fully understand here,
you defined a array with size of 2, but looks like you only use the first element because 
const int kResInfo = 0;

so why didn't you just define a variable instead of an array?

Reply:
1 object may have 2 explicit bindings simultaneously.
`ResAndSemanticInfo` only stores the explicit binding information of 1 explicit binding.
`GLSLInputAttachmentIndexLayoutAttribute` may appear along side any other binding attribute. 
There is a minimum of 1 binding, max of 2 bindings, instead of allocating a List, an array on the stack is faster.

Reply:
Yes, I understand.
But my question is where are you actually using `info[kSubpassResInfo]`?

Reply:
`info[kSubpassResInfo]` is used when generating parameter bindings
```
        for (int i = 0; i < kMaxResCount; i++)
        {
            auto* resInfoItem = info[i].resInfo;
            auto& semanticInfo = info[i].semanticInfo;
```